### PR TITLE
Improve documentation around .NET Framework and gRPC-Web in the client

### PR DESCRIPTION
**Pull request by JamesNK**
Fixes https://github.com/grpc/grpc-dotnet/issues/2197

@jtattermusch Is this what you're looking for?

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/grpc/netstandard.md](https://github.com/dotnet/AspNetCore.Docs/blob/32e76a58222347e7dd61c9b5f93277275adc9ecd/aspnetcore/grpc/netstandard.md) | [Use gRPC client with .NET Standard 2.0](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/netstandard?branch=pr-en-us-30033) |
| [aspnetcore/grpc/supported-platforms.md](https://github.com/dotnet/AspNetCore.Docs/blob/32e76a58222347e7dd61c9b5f93277275adc9ecd/aspnetcore/grpc/supported-platforms.md) | [gRPC on .NET supported platforms](https://review.learn.microsoft.com/en-us/aspnet/core/grpc/supported-platforms?branch=pr-en-us-30033) |

### CODE REVIEW

File: aspnetcore/grpc/netstandard.md [hunk 1]
```diff
@@ -69,6 +69,8 @@ For more information, see [Configure gRPC-Web with the .NET gRPC client](xref:gr
 Requirements and restrictions to using `WinHttpHandler`:
 
 * Windows 11 or later, Windows Server 2022 or later.
+  * gRPC client is fully supported on Windows 11 or later.
+  * gRPC client is partially supported on Windows Server 2022. Unary and server streaming methods are supported. Client and bidirectional streaming methods are not supported.
```
Comment (Rick-Anderson) [inline aspnetcore/grpc/netstandard.md:73]: Pick one
```suggestion
  * gRPC client is partially supported on Windows Server 2022. Unary and server streaming methods are supported. Client and bidirectional streaming methods aren't supported.
```

```suggestion
  * gRPC client is partially supported on Windows Server 2022. Unary and server streaming methods are supported. Client and bidirectional streaming methods are **_not_** supported.
```

**Review by Rick-Anderson — APPROVED** [review]

**Comment by jtattermusch** [comment]
@JamesNK Yes, thanks for updating the docs.

File: aspnetcore/grpc/netstandard.md [hunk 2]
```diff
@@ -69,6 +69,8 @@ For more information, see [Configure gRPC-Web with the .NET gRPC client](xref:gr
 Requirements and restrictions to using `WinHttpHandler`:
 
 * Windows 11 or later, Windows Server 2022 or later.
+  * gRPC client is fully supported on Windows 11 or later.
+  * gRPC client is partially supported on Windows Server 2022. Unary and server streaming methods are supported. Client and bidirectional streaming methods are **_not_** supported.
```
Comment (jtattermusch) [inline aspnetcore/grpc/netstandard.md:73]: qq: is the mode of failure for "not supported" functionality documented somewhere else than the code itself?

Basically, 
1. What happens when you try any gRPC call on platform that doesn't support WinHttpHandler at all.
2. what happens when you attempt client streaming or bi streaming call on a system where that's not supported.
3. what happens when you try non-TLS with WinHttpHandler

I don't think this needs to be documented here specifically, but may well be worth mentioning somewhere else.

[the remark this answers is not in the document]
Comment (JamesNK) [reply aspnetcore/grpc/netstandard.md:73]: Probably the next version of Windows Server. 2024?

Comment (JamesNK) [reply aspnetcore/grpc/netstandard.md:73]: > What happens when you try any gRPC call on platform that doesn't support WinHttpHandler at all.

I think WinHttpHandler throws a platform not supported exception that Windows is required.

> what happens when you attempt client streaming or bi streaming call on a system where that's not supported.
> what happens when you try non-TLS with WinHttpHandler

I recall that in both these cases WinHttpHandler silently falls back to sending an HTTP/1.1 request which then fails.

**Review by wadepickett — APPROVED** [review]
Looks good with currently made.

**Pull request merged**